From 63ceb647055e62842a621e7a819af461f348943f Mon Sep 17 00:00:00 2001
From: Steve Wainstead <wainstead@gmail.com>
Date: Sun, 28 Jun 2009 22:55:32 +0000
Subject: [PATCH 020/223] after macro expansion, g++ complained that:

g++ -O  -c -o network.o network.c
network.c: In function 'Var network_connection_options(network_handle, Var)':
network.c:29: error: invalid conversion from 'int' to 'var_type'
make: *** [network.o] Error 1

so I've commented out the original macros and replaced it with the
expanded code, as found via g++ -E.
---
 server/network.c |   46 +++++++++++++++++++++++++++++++++++++++++++++-
 1 file changed, 45 insertions(+), 1 deletion(-)

diff --git a/server/network.c b/server/network.c
index 9f741b5..c7cea34 100644
--- a/server/network.c
+++ b/server/network.c
@@ -26,6 +26,50 @@
 Var
 network_connection_options(network_handle nh, Var list)
 {
+  do { 
+    { 
+      Var pair = new_list(2);
+      pair.v.list[1].type = (var_type)(_TYPE_STR | 0x80);
+      pair.v.list[1].v.str = str_dup("client-echo");
+      pair.v.list[2].type = (TYPE_INT);
+      pair.v.list[2].v.num = (((nhandle *)(nh).ptr)->client_echo);
+      (list) = listappend((list), pair);
+    } return (list);
+  } while (0);
+}
+
+int
+network_connection_option(network_handle nh, const char *option, Var * value)
+{
+  do { if (!mystrcasecmp((option), "client-echo")) { (value)->type = (TYPE_INT);
+      (value)->v.num = (((nhandle *)(nh).ptr)->client_echo);
+      return 1;
+    } return 0;
+  } while (0);
+}
+
+int
+network_set_connection_option(network_handle nh, const char *option, Var value)
+{
+  do { if (!mystrcasecmp((option), "client-echo")) { network_set_client_echo((nh), is_true((value)));;
+      return 1;
+    } return 0;
+  } while (0);
+}
+
+/*
+ * after macro expansion, g++ complained that:
+
+g++ -O  -c -o network.o network.c
+network.c: In function 'Var network_connection_options(network_handle, Var)':
+network.c:29: error: invalid conversion from 'int' to 'var_type'
+make: *** [network.o] Error 1
+
+* so I'm commenting this out and placing the macro-expanded copy above.
+
+Var
+network_connection_options(network_handle nh, Var list)
+{
     CONNECTION_OPTION_LIST(NETWORK_CO_TABLE, nh, list);
 }
 
@@ -40,7 +84,7 @@ network_set_connection_option(network_handle nh, const char *option, Var value)
 {
     CONNECTION_OPTION_SET(NETWORK_CO_TABLE, nh, option, value);
 }
-
+*/
 
 /* 
  * $Log: not supported by cvs2svn $
-- 
1.7.9.5

